src/core/viz/expressions/basic/array.js

```diff
@@ -0,0 +1,59 @@
+import BaseExpression from '../base';
+import { checkExpression, implicitCast } from '../utils';
+
+/**
+ * Wrapper around arrays. Explicit usage is unnecessary since CARTO VL will wrap implicitly all arrays using this function.
+ *
+ * @param {Number[]|Category[]|Color[]|Date[]} elements
+ * @returns {Array}
+ *
+ * @memberof carto.expressions
+ * @name array
+ * @function
+ * @IGNOREapi
+ */
+export default class BaseArray extends BaseExpression {
+    constructor(elems) {
+        elems = elems || [];
+        if (!Array.isArray(elems)) {
+            elems = [elems];
+        }
+        elems = elems.map(implicitCast);
+        if (!elems.length) {
+            throw new Error('array(): invalid parameters: must receive at least one argument');
+        }
+        let type = '';
+        for (let elem of elems) {
+            type = elem.type;
+            if (elem.type != undefined) {
+                break;
+            }
+        }
+        if (['number', 'category', 'color', 'time', undefined].indexOf(type) == -1) {
+            throw new Error(`array(): invalid parameters type: ${type}`);
+        }
+        elems.map((item, index) => {
+            checkExpression('array', `item[${index}]`, index, item);
+            if (item.type != type && item.type != undefined) {
+                throw new Error('array(): invalid parameters, invalid argument type combination');
+            }
+        });
+        super({});
+        this.type = type + '-array';
+        this.expr = elems;
```

Comment:
`expr` is not a descriptive name, could we use `elems`?

Reply:
Yes. I just kept the old name, but elems is better.